# Descriptive Statistics Assignment

In [ ]:
import pandas as pd

data_url = "http://data.insideairbnb.com/united-states/ny/new-york-city/2020-06-08/visualisations/listings.csv"

### Read the aggregated AirBnB listings of New York (using `data_url`) into a Pandas data frame.

In [ ]:
data = pd.read_csv(data_url)

In [ ]:
data.head(5)

id  ... availability_365
0  2060  ...              365
1  2595  ...              335
2  3831  ...              276
3  5099  ...                0
4  5121  ...              365

[5 rows x 16 columns]

### Print the data type for each column.

In [ ]:
data.dtypes

id                                  int64
name                               object
host_id                             int64
host_name                          object
neighbourhood_group                object
neighbourhood                      object
latitude                          float64
longitude                         float64
room_type                          object
price                               int64
minimum_nights                      int64
number_of_reviews                   int64
last_review                        object
reviews_per_month                 float64
calculated_host_listings_count      int64
availability_365                    int64
dtype: object

### Generate a basic set of descriptive statistics for the numeric fields in the data set.

In [ ]:
desc = data.describe()

### Calculate the following for each numeric field in the data set and add them to the data frame containing the basic set of descriptive statistics.

- Variance
- Mean Absolute Deviation
- Range
- Interquartile Range

In [ ]:
data.var()

id                                1.829752e+14
host_id                           9.776457e+15
latitude                          2.989206e-03
longitude                         2.260709e-03
price                             1.758228e+05
minimum_nights                    4.828933e+02
number_of_reviews                 2.327659e+03
reviews_per_month                 1.809598e+00
calculated_host_listings_count    6.495002e+02
availability_365                  2.027247e+04
dtype: float64

In [ ]:
data.mad()

id                                1.181604e+07
host_id                           8.179277e+07
latitude                          4.336345e-02
longitude                         3.274414e-02
price                             1.090032e+02
minimum_nights                    8.992644e+00
number_of_reviews                 2.906928e+01
reviews_per_month                 9.489338e-01
calculated_host_listings_count    8.506215e+00
availability_365                  1.265533e+02
dtype: float64

In [ ]:
max_min = desc.loc[['max', 'min']]

In [ ]:
for col in max_min:
    max_col = max_min[col].loc['max']
    min_col = max_min[col].loc['min']
    print (f'range in {col} = {max_col - min_col}')

range in id = 43701299.0
range in host_id = 349080382.0
range in latitude = 0.4118999999999957
range in longitude = 0.5278500000000008
range in price = 10000.0
range in minimum_nights = 1249.0
range in number_of_reviews = 746.0
range in reviews_per_month = 53.79
range in calculated_host_listings_count = 279.0
range in availability_365 = 365.0


### Generate a basic set of descriptive statistics for the categorical fields in the data set.

In [ ]:
data.describe(include="object").T

count unique                  top   freq
name                 49512  48247  Home away from home     21
host_name            49524  11572              Michael    416
neighbourhood_group  49530      5            Manhattan  21963
neighbourhood        49530    222         Williamsburg   3752
room_type            49530      4      Entire home/apt  25716
last_review          38211   2077           2020-01-01    940

### Calculate the Pearson correlation coefficients between price, minimum_nights, reviews_per_month, calculated_host_listings_count and availability_365

In [ ]:
new_data = data[['price', 'minimum_nights', 'reviews_per_month', 'calculated_host_listings_count', 'availability_365']]
new_data.corr('pearson')

price  ...  availability_365
price                           1.000000  ...          0.055788
minimum_nights                  0.015923  ...          0.148435
reviews_per_month              -0.011623  ...          0.213093
calculated_host_listings_count  0.004730  ...          0.173181
availability_365                0.055788  ...          1.000000

[5 rows x 5 columns]

### Group the data by neighbourhood and calculate the mean and standard deviation of revenue for each neighbourhood.

In [ ]:
aggs = {'price' : ['mean', 'std']}
new_df = data.groupby(['neighbourhood_group']).agg(aggs).reset_index()
new_df

neighbourhood_group       price            
                             mean         std
0               Bronx   90.176127   98.569331
1            Brooklyn  125.056194  204.835722
2           Manhattan  218.855166  583.519236
3              Queens   99.745056  195.717338
4       Staten Island  116.908108  241.447121

### Create a new mimimum revenue index variable that combines the price, minimum nights, calculate host listings count into a single value for each dwelling. Then sort the data frame by that value to see which dwelling has the most financial strength. 



In [ ]:
data['revenue_index'] = (data['price']) * (data['minimum_nights']) * (data['calculated_host_listings_count'])

data.sort_values('revenue_index', ascending= False).head(5).reset_index() 

index        id  ... availability_365  revenue_index
0  14754  13443640  ...              184       10956000
1  31916  30389115  ...              145        8467200
2   9961   8810949  ...              365        7768800
3  33929  32399523  ...              251        6804000
4  31992  30394518  ...              318        5275200

[5 rows x 18 columns]

### Group the data by neighbourhood_group and calculate the average minimum revenue index for each neighbourhood_group.

In [ ]:
index_neighborhood = data.groupby(['revenue_index']).sum()
index_neighborhood.sort_values('revenue_index', ascending= False).head(5).reset_index() 

revenue_index        id  ...  calculated_host_listings_count  availability_365
0       10956000  13443640  ...                              83               184
1        8467200  30389115  ...                             280               145
2        7768800   8810949  ...                              83               365
3        6804000  32399523  ...                             280               251
4        5275200  30394518  ...                             280               318

[5 rows x 11 columns]

1. Identify 3 scenarios in your everyday life where you use probability.

Will our favorite team win the next game?

Depending on what day of the week it is and what time it is, there will be a certain amount of traffic.

What is the chance of rain based on how many weeks per year it rains?

2. Using the formula you learned to calculate probabilities, approximate the probabilities of the events in each of those 3 scenarios.

a) For the question: "Will our favorite team win the next game?"

Our team has won 7 out of 10 games, so their chance of winning is 7/10 or 70%

The other team has won 8/10 so their probability of winning is 80% (or 20% of losing)

P(AB) = P(A|B) * P(B)
A = our team wins
B = the other team loses

P(AB) = game ends in a draw

P(A) = .70
P(B) = .20

Joint probability  = .14 = 14%

this means that there is a 14% chance that the game will end in a draw

b) for the question: 'What is the probability of getting stuck in traffic on the way to work?'

heavy traffic = 5/7 = .71

light traffic = 2/7 = .29

joint probability = .2 = 20 %

c) for the question 'What is the chance of getting rain and thunder?'

chance of rain = 1/52 = 1.9%

chance of thunder is 2/7 = 29%

joint probability = .005 = 0.5%


3. Think about what potential dependencies the events in each of your 3 scenarios have. Calculate the joint probabilities for the events and their dependencies.

a) the dependency for our team to win is if the other team loses

b) the dependency for getting stuck in traffic is if it is a weekday

c) the dependency for rain and thunder is if the moisture level is over 50%

4. Calculate the conditional probabilities for the events given their dependencies.

a) the joint probability is 14% and the dependency is 20%. The conditional probabilty is 70%

b) joint probabilty = 20%. Dependency = 5/7 = 71%.
Conditional probability = 28.5%

c) joint probability = 0.5%, dependency = 3%
conditional probability = 17%